File: Ruby/Authored/Hippo.rb
class Game
  def initialize(board)
    @amount = 0
    @board = board
    @done = []
    (0..@board.length-1).each do |x|
      temp = []
      (0..@board.length-1).each do |y|
        temp << false
      end
      @done << temp
    end
  end

  def play()
    (0..@board.length-1).each do |x|
      (0..@board.length-1).each do |y|
        checkmysquare(x,y) if @board[x][y] == 1 && !@done[x][y]
      end
    end
    @amount
  end

  def checkmysquare(x,y)
    @amount += 1
    temp = [[x,y]]
    num = 0
    while (num < temp.length) do
      x = temp[num][0]
      y = temp[num][1]
      @done[x][y] = true
      if x != 0
        if @board[x-1][y] == 1 && !@done[x-1][y]
          temp << [x-1,y]
          @done[x-1][y] = true
        end
      end
      if x != @board.length - 1
        if @board[x+1][y] == 1 && !@done[x+1][y]
          temp << [x+1,y]
          @done[x+1][y] = true
        end
      end
      if y != 0
        if @board[x][y-1] == 1 && !@done[x][y-1]
          temp << [x,y-1]
          @done[x][y-1] = true
        end
      end
      if y != @board.length - 1
        if @board[x][y+1] == 1 && !@done[x][y+1]
          temp << [x,y+1]
          @done[x][y+1] = true
        end
      end
      num += 1
    end
  end
end

def randomBoard()
  srand
  board = []
  boardSize = rand(45)+5
  blocks = rand(boardSize*boardSize)
  p boardSize
  p blocks
  (0..boardSize).each do |x|
    temp = []
    (0..boardSize).each do |y|
      temp << 0
    end
    board << temp
  end
  (0..blocks).each do |x|
    x = rand(boardSize)
    y = rand(boardSize)
    board[x][y] = 1
  end
  board
end

File: CSharp/Authored/Hippo.cs
using System.Collections.Generic;

namespace CodeWars
{
    public class Game
    {
        private int[,] board = new int[,] { };
        private bool[,] done = new bool[,] { };
        private int amount = 0;

        public Game(int[,] board2)
        {
            board = board2;
            int boardSize = board.GetUpperBound(0) + 1;
            done = new bool[boardSize, boardSize];
            for (int x=0; x < boardSize; x++)
            {
                for(int y=0; y< boardSize; y++)
                {
                    done[x, y] = false;
                }
            }
        }

        public int play()
        {
            int boardSize = board.GetUpperBound(0)+1;
            for (int x = 0; x < boardSize; x++)
            {
                for (int y = 0; y < boardSize; y++)
                {
                    if (board[x,y] == 1 && !done[x, y]) { checkmysquare(x, y); }
                }
            }
            return amount;
        }

        public void checkmysquare(int a, int b)
        {
            amount++;
            int boardSize = board.GetUpperBound(0);
            List<int> xx = new List<int> { a };
            List<int> yy = new List<int> { b };
            int num = 0;
            while (num < xx.Count)
            {
                int x = xx[num];
                int y = yy[num];
                done[x, y] = true;
                if (x != 0) {
                    if (board[x - 1, y] == 1 && !done[x - 1, y]) {
                        xx.Add(x - 1);
                        yy.Add(y);
                        done[x - 1, y] = true;
                    }
                }
                if (x != boardSize) {
                    if (board[x + 1, y] == 1 && !done[x + 1, y]) {
                        xx.Add(x + 1);
                        yy.Add(y);
                        done[x + 1, y] = true;
                    }
                }
                if (y != 0) {
                    if (board[x, y - 1] == 1 && !done[x, y - 1]) {
                        xx.Add(x);
                        yy.Add(y - 1);
                        done[x, y - 1] = true;
                    }
                }
                if (y != boardSize)
                {
                    if (board[x, y + 1] == 1 && !done[x, y + 1])
                    {
                        xx.Add(x);
                        yy.Add(y + 1);
                        done[x, y + 1] = true;
                    }
                }
                num++;
            }

        }

    }
}